web/controllers/survey_controller.ex

```diff
@@ -1,14 +1,39 @@
 defmodule Ask.SurveyController do
   use Ask.Web, :api_controller
+  use Ask.Web, :append_assigns_to_action
+
+  import Survey.Helper
 
   alias Ask.{Project, Folder, Survey, Questionnaire, Logger, RespondentGroup, Respondent, Channel, ShortLink, ActivityLog, RetriesHistogram, ScheduleError, ConflictError}
   alias Ask.Runtime.{Session, SurveyAction}
   alias Ecto.Multi
 
-  def index(conn, %{"project_id" => project_id} = params) do
-    project = conn
-    |> load_project(project_id)
-
+  plug :assign_project when action in [
+    :index,
+    :show,
+    :stats,
+    :retries_histograms,
+    :simulation_initial_state,
+    :simulation_status,
+    :stop_simulation,
+  ]
+  plug :assign_project_for_change when action in [
+    :create,
+    :update,
+    :set_folder_id,
+    :set_name,
+    :set_description,
+    :delete,
+    :launch,
+    :simulate_questionanire,
+    :create_link,
+    :delete_link,
+    :refresh_link,
+    :stop,
+  ]
+  plug :assign_project_for_owner when action in [:update_locked_status]
```

Comment:
We could extract:
- the 4 simulation methods into a SurveySimulationController
- the 3 link methods into a SurveyLinkController